### DISCUSSION TITLE
YantraJS is thread-safe?

### DISCUSSION THREAD

**Discussion by kitsunoff** [yantrajs/yantra#55] [Q&A] [answered]
Hi ackava
I am currently start writing a desktop ui framework, that use yantrajs under the hood
And i have question
Can i safety change JSObject/JSValue properties from other thread? 
(for example in ui thread button name is changed and needed change name property in "button-proxy" in js thread)

**Comment by ackava** [comment, marked as answer]
When you create JSContext from UI Thread, all operations will be performed on UI Thread, just like how JavaScript is executed on browser. `setTimeout`, `setInterval` and all proxy resolutions will execute on the UI Thread.

JSContext isn't threadsafe, as JavaScript was itself designed to run on single thread. If you modify JSObject from other threads, it will lead to inconsistent results. But you can always queue your changes on the main thread.

**Reply by kitsunoff** [reply]
Thank you so much
